src/components/strategies/common/d3Chart/D3PriceHistory.tsx

```diff
@@ -101,33 +97,44 @@ const useZoom = (
         if (e.transform.k === k) chartArea.style('cursor', 'grab');
         setTransform(e.transform);
       })
-      .on('end', () => chartArea.style('cursor', ''));
-    selection.call(zoomHandler.current);
-  }, [chartArea, extent.translate, extent.zoom, selection]);
+      .on('end', () => {
+        chartArea.style('cursor', '');
+      });
+    return handler;
+  }, [behavior, data.length, dms.width]);
 
-  const zoomRange = (from: string, days: number) => {
-    const baseXScale = (() => {
-      if (behavior === 'normal') {
-        return scaleBand()
-          .domain(data.map((d) => d.date.toString()))
-          .range([0, dms.boundedWidth])
-          .paddingInner(0.5);
-      } else {
-        return scaleBand()
-          .domain(getExtendedRange(data.map((d) => d.date)))
-          .range([dms.boundedWidth * -0.5, dms.boundedWidth * 1.5])
-          .paddingInner(0.5);
-      }
-    })();
+  useEffect(() => {
+    select<SVGSVGElement, unknown>('#interactive-chart').call(zoomHandler);
+  }, [zoomHandler]);
 
-    const scale = data.length / days;
-    const translateX = baseXScale(from)!;
-    const transition = selection.transition().duration(500);
-    const transform = zoomIdentity.scale(scale).translate(-1 * translateX, 0);
-    zoomHandler.current?.transform(transition, transform);
-  };
+  const zoomRange = useCallback(
+    (from: string, days: number, duration = 500) => {
+      const baseXScale = (() => {
+        if (behavior === 'normal') {
+          return scaleBand()
+            .domain(data.map((d) => d.date.toString()))
+            .range([0, dms.boundedWidth])
+            .paddingInner(0.5);
+        } else {
+          return scaleBand()
+            .domain(getExtendedRange(data.map((d) => d.date)))
+            .range([dms.boundedWidth * -0.5, dms.boundedWidth * 1.5])
+            .paddingInner(0.5);
+        }
+      })();
 
-  return { transform, zoomRange };
+      const selection = select<SVGSVGElement, unknown>('#interactive-chart');
+      const scale = data.length / days;
+      const translateX = baseXScale(from)!;
+      const transition = selection.transition().duration(duration);
+      const transform = zoomIdentity.scale(scale).translate(-1 * translateX, 0);
+      zoomHandler?.transform(transition, transform);
+      return new Promise((res) => setTimeout(res, duration));
+    },
+    [behavior, data, dms.boundedWidth, zoomHandler]
```

Comment:
```suggestion
    [behavior, data.length, dms.boundedWidth, zoomHandler]
```

Reply:
I cannot because I need to map over the dates:
```js
.domain(data.map((d) => d.date.toString()))
```

Technically I could (and I should I guess) because I know the array will always be the same... But so far I don't see any rendering issue right now, so I would rather not play with these yet.